compiler/inference/helpers.py

```diff
@@ -1,11 +1,44 @@
 from collections.abc import Generator
 
-from xdsl.dialects import scf
+from xdsl.dialects import func, llvm, scf
 from xdsl.ir import Block, BlockArgument, Operation, OpResult, Region, SSAValue
 
 from compiler.dialects import accfg
 
 
+def has_accfg_effects(op: Operation) -> bool:
+    """
+    Checks if an operation effects state managed by the accfg dialect, according to the following criteria:
+
+    - If provided, an attribute named `accfg_effects` of either of the two types will overwrite inference.
+      These attributes will need to be added by the provider of the IR, as only they know which functions may affect
```

Comment:
Can we as "providers of the IR" just add attributes to any operation? Will this not introduce parsing errors?
E.g. A `memref.subview` operation with a `accfg_effect` attribute is not a valid `memref.subview` operation anymore, is it?

Reply:
An operation can have arbitrary attributes, they are meant to be used like this. Properties are "fixed", as in there is a fixed set of them per operation, but you can set arbitrary attributes on any operation.